src/map/GMap.js

```diff
@@ -238,6 +239,25 @@
         this._prevCenterLat = this.centerLat();
         this._prevCenterLong = this.centerLong();
         this._prevZoom = this.zoom();
+        
+        if (this.mapDrawingTools()) {
```

Comment:
In the most recent commit, I've kept the drawing tools initialization in the enter function, but I've moved setting the map object on the drawing tools object into the update method, and this allows for toggling the drawing functionality on/off.  I did this so that the update function effectively enables/disables the drawing tools, but doesn't have to concern itself with initializing or tearing down the drawing tools object.

Additionally, added a published property which is just a running array of the objects drawn on the map.